### PR TITLE
Migration: fix RemoteRepository - Project data migration

### PR DESCRIPTION
**Pull request by humitos**
Calling `RemoteRepository.save()` does not save the relation in `Project`.
This commit changes it to call `Project.save()` instead.

Calling `.save()` over a table where its schema is also being modified produces
a PostgreSQL error:

```
database_1 | 2021-06-17 12:05:45.877 UTC [193] ERROR: cannot ALTER TABLE "projects_project" because it has pending trigger events
database_1 | 2021-06-17 12:05:45.877 UTC [193] STATEMENT: SET CONSTRAINTS "oauth_remotereposito_project_id_246c363d_fk_projects_" IMMEDIATE; ALTER TABLE "oauth_remoterepository_2020" DROP CONSTRAINT "oauth_remotereposito_project_id_246c363d_fk_projects_"
```

See the discussion at https://github.com/readthedocs/readthedocs.org/pull/8259/files#r653500791

I found a solution following this SO answer that consist on splitting the
migration into 1) schema migration and 2) data migration: https://stackoverflow.com/a/12838113

I tested this locally with my database with `Projects` connected to
`RemoteRepositries` and kept the relations after the migration was applied.

### CODE REVIEW

File: readthedocs/projects/migrations/0077_remote_repository_data_migration.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Generated by Django 2.2.24 on 2021-06-17 12:14
+
+from django.db import migrations
+
+
+def migrate_data(apps, schema_editor):
+    RemoteRepository = apps.get_model('oauth', 'RemoteRepository')
+    queryset = RemoteRepository.objects.filter(project__isnull=False).select_related('project')
```
Comment (saadmk11) [inline readthedocs/projects/migrations/0077_remote_repository_data_migration.py:8]:
```suggestion
    queryset = RemoteRepository.objects.filter(project__isnull=False).select_related('project').only('pk', 'project')
```

Maybe we can use `only()` here.

**Review by saadmk11 — APPROVED** [review]
Changes look good to me! 🎉

File: readthedocs/oauth/migrations/0014_remove_remoterepository_project.py [hunk 2]
```diff
@@ -3,23 +3,14 @@
...
 
     dependencies = [
         ('oauth', '0013_create_new_table_for_remote_repository_normalization'),
-        ('projects', '0076_project_remote_repository'),
+        ('projects', '0077_remote_repository_data_migration'),
     ]
 
     operations = [
-        migrations.RunPython(migrate_data),
         migrations.RemoveField(
```
Comment (stsewd) [inline readthedocs/oauth/migrations/0014_remove_remoterepository_project.py:14]: We should add a note on the deploy project to run these one by one, so we can check the migration was done correctly before we delete this field. `migrate projects 0077` and then `migration oauth 0014`

File: readthedocs/projects/migrations/0077_remote_repository_data_migration.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Generated by Django 2.2.24 on 2021-06-17 12:14
+
+from django.db import migrations
+
+
+def migrate_data(apps, schema_editor):
+    RemoteRepository = apps.get_model('oauth', 'RemoteRepository')
+    queryset = RemoteRepository.objects.filter(project__isnull=False).select_related('project').only('pk', 'project')
+    for rr in queryset.iterator():
+        rr.project.remote_repository_id = rr.pk
```
[the remark this answers is not in the document]
Comment (humitos) [reply readthedocs/projects/migrations/0077_remote_repository_data_migration.py:10]: No, we need to save the `id` from the remote repository into the project's row.

This line does:

- finds the associated project from a remote repository
- in that project, it copies the remote repository ids into the new project's ForeignKey field (`remote_repository`) --which can be accessed via `remote_repository_id`

File: readthedocs/oauth/migrations/0014_remove_remoterepository_project.py (hunk 2 above)
Comment (humitos) [reply readthedocs/oauth/migrations/0014_remove_remoterepository_project.py:14]: Yeah! I added a note with a warning and all the steps required already! 👍🏼

**Review by stsewd — APPROVED** [review]

**Pull request merged**